src/test/rustdoc/doctest-keyword.rs

```diff
@@ -0,0 +1,6 @@
+#![crate_name = "mod"]
+//! ```
+//! // NOTE: requires that the literal string 'mod' appears in the doctest for
+//! // the bug to appear
+//! assert_eq!(1, 1);
+//! ```
```

Comment:
I don't know how to test this. Currently the test does nothing because it isn't running rustdoc with `--test`; if I add `// compile-flags: --test` it errors regardless of the fix because the aux metadata isn't built:

```
running 1 test
test src/test/rustdoc/doctest-keyword.rs - (line 3) ... FAILED

failures:

---- src/test/rustdoc/doctest-keyword.rs - (line 3) stdout ----
error[E0463]: can't find crate for `mod`
 --> src/test/rustdoc/doctest-keyword.rs:4:1
  |
4 | extern crate r#mod;
  | ^^^^^^^^^^^^^^^^^^^ can't find crate
```

I can't seem to find any other tests for this? @GuillaumeGomez do you have ideas?

Reply:
You can use `aux-build` to achieve that I think, no?

Reply:
Hmm, I'm not sure how. Normally `aux-build` looks in the `auxiliary` directory, so I tried using `// aux-build:../doctest-keyword.rs` to trick it into building the current file. But it still seems to be breaking :/

<details>

```
running 6 tests
iiiiiF
failures:

---- [rustdoc] rustdoc/doctest-keyword.rs stdout ----


executing "/home/joshua/rustc/build/x86_64-unknown-linux-gnu/stage1/bin/rustc" "/home/joshua/rustc/src/test/rustdoc/auxiliary/../doctest-keyword.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "-C" "prefer-dynamic" "--out-dir" "/home/joshua/rustc/build/x86_64-unknown-linux-gnu/test/rustdoc/doctest-keyword/auxiliary" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/home/joshua/rustc/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--test" "--crate-type" "dylib" "-L" "/home/joshua/rustc/build/x86_64-unknown-linux-gnu/test/rustdoc/doctest-keyword/auxiliary"
------stdout------------------------------

------stderr------------------------------

------------------------------------------
executing "/home/joshua/rustc/build/x86_64-unknown-linux-gnu/stage1/bin/rustdoc" "-L" "/home/joshua/rustc/build/x86_64-unknown-linux-gnu/stage1/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-L" "/home/joshua/rustc/build/x86_64-unknown-linux-gnu/test/rustdoc/doctest-keyword/auxiliary" "-o" "/home/joshua/rustc/build/x86_64-unknown-linux-gnu/test/rustdoc/doctest-keyword" "/home/joshua/rustc/src/test/rustdoc/doctest-keyword.rs" "--test"
------stdout------------------------------

running 1 test
test src/test/rustdoc/doctest-keyword.rs - (line 7) ... FAILED

failures:

---- src/test/rustdoc/doctest-keyword.rs - (line 7) stdout ----
error[E0463]: can't find crate for `mod`
 --> src/test/rustdoc/doctest-keyword.rs:8:1
  |
4 | extern crate r#mod;
  | ^^^^^^^^^^^^^^^^^^^ can't find crate

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
Couldn't compile the test.

failures:
    src/test/rustdoc/doctest-keyword.rs - (line 7)

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.02s
```

</details>

I did confirm that the aux file exists, but it doesn't seem to be picking up on it for some reason.

```
$ file /home/joshua/rustc/build/x86_64-unknown-linux-gnu/test/rustdoc/doctest-keyword/auxiliary/mod
/home/joshua/rustc/build/x86_64-unknown-linux-gnu/test/rustdoc/doctest-keyword/auxiliary/mod: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=df3038c014cbf237c458cb19f7571c62aa8b96f9, not stripped
```

@Mark-Simulacrum do you have ideas maybe?

Reply:
The problem here might coming from the `.rs` file generated by rustdoc containing the test, no?

Reply:
I think it's unlikely this will regress, since someone would have to change it intentionally. Do you think it makes sense to add the fix before figuring out how to test it?